src/error.rs

```diff
@@ -166,6 +166,20 @@ pub enum SimpleReason<I, S> {
     Custom(String),
 }
 
+impl<I, S> SimpleReason<I, S> {
+    const DEFAULT_DISPLAY_UNEXPECTED: &'static str = "unexpected value";
```

Comment:
I think it's probably find to just have this inline within the implementation of `Display`. I would probably change it to `"unexpected input"` though, in order to retain consistency with other parts of the codebase.

Reply:
> I think it's probably find to just have this inline within the implementation of Display.

I'd prefer to have this as a constant here in order to avoid [magic strings](https://en.wikipedia.org/wiki/Magic_string). But if you wish I can put it inline into the implementation of the `Display` trait.

> I would probably change it to "unexpected input" though, in order to retain consistency with other parts of the codebase.

Understandable, I'll change that ~~tomorrow~~ today, if I've slept.

Reply:
> I'd prefer to have this as a constant here in order to avoid [magic strings](https://en.wikipedia.org/wiki/Magic_string)

I'm not sure this is the same as a magic string. The text doesn't have any particularly special meaning outside of the `Display` impl itself, does not result in unusual behaviour, and the constant is neither configurable not publicly visible anyway. There's also another similar string already in the `Display` impl already.

Reply:
done